<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to Phylogenetic Models of Discrete Biogeography • SystematicsClass</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Introduction to Phylogenetic Models of Discrete Biogeography">
<meta property="og:description" content="SystematicsClass">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">SystematicsClass</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/00_Basics_of_R/index.html">R Reminder</a>
    </li>
    <li>
      <a href="../../articles/01_homology_assignment/index.html">Morphological Homology Assignment</a>
    </li>
    <li>
      <a href="../../articles/02_molecular_homology/index.html">Molecular Homology Assignment</a>
    </li>
    <li>
      <a href="../../articles/03_massive_homology/index.html">Genome-Scale Homology</a>
    </li>
    <li>
      <a href="../../articles/04_ParsimonyAndModels/index.html">PAUP, parsimony, and nucletoide substitution models</a>
    </li>
    <li>
      <a href="../../articles/05_CTMC/index.html">Nucleotide substitution models</a>
    </li>
    <li>
      <a href="../../articles/05_models/index.html">Nucleotide substitution models</a>
    </li>
    <li>
      <a href="../../articles/06_Bayes_and_MCMC/index.html">Introduction to MCMC using RevBayes</a>
    </li>
    <li>
      <a href="../../articles/07_molecular_data/index.html">Nucleotide substitution models</a>
    </li>
    <li>
      <a href="../../articles/08_Discrete_Morphology/index.html">Discrete morphology - Models and Tree Inference</a>
    </li>
    <li>
      <a href="../../articles/09_Partitioned_data/index.html">Partitioned data analysis</a>
    </li>
    <li>
      <a href="../../articles/10_Molecular_Clock/index.html">Tripartite Model 2 - Clock Models for Character Data</a>
    </li>
    <li>
      <a href="../../articles/11_FBD/RB_Total_Evidence_Tutorial.html">RevBayes Total Evidence Tree Estimation</a>
    </li>
    <li>
      <a href="../../articles/13_Comparative_Methods/index.html">Phylogenetic Comparative Methods for Paleobiology</a>
    </li>
    <li>
      <a href="../../articles/14_biogeography/index.html">Introduction to Phylogenetic Models of Discrete Biogeography</a>
    </li>
    <li>
      <a href="../../articles/15_Final/index.html">FinalExam</a>
    </li>
    <li>
      <a href="../../articles/LCalc/LikelihoodCalc.html">LikelihoodCalc</a>
    </li>
    <li>
      <a href="../../articles/ModelsHomework/index.html">Models Homework</a>
    </li>
    <li>
      <a href="../../articles/PaupHomework/index.html">PAUPHomework</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to Phylogenetic Models of Discrete Biogeography</h1>
            <h3 data-toc-skip class="subtitle">Overview of the Dispersal-Extirpation-Cladogenesis (DEC) model</h3>
            
      
      
      <div class="hidden name"><code>index.Rmd</code></div>

    </div>

    
    
<p>{% section Introduction | introduction %}</p>
<p>Many fundamental evolutionary processes, such as adaptation, speciation, and extinction, operate in a spatial context. When the historical aspect of this spatial context cannot be observed directly, as is often the case, biogeographic inference may be applied to estimate ancestral species ranges. This works by leveraging phylogenetic, molecular, and geographical information to model species distributions as the outcome of biogeographic processes. How to best model these processes requires special consideration, such as how ranges are inherited following speciation events, how geological events might influence dispersal rates, and what factors affect rates of dispersal and extirpation. A major technical challenge of modeling range evolution is how to translate these natural processes into stochastic processes that remain tractable for inference. This tutorial provides a brief background in some of these models, then describes how to perform Bayesian inference of historical biogeography using RevBayes.</p>
<!-- {% section Getting Set Up | setup %}

This first exercise will not require you to use any data. However, it may still be
useful to work within a single directory.

> Create a new directory on your computer called `RB_biogeo_tutorial`.
> 
> Navigate to the `RB_biogeo_tutorial` directory and execute the `rb` binary. 
> One option for doing this is to move the `rb` executable to the `RB_biogeo_tutorial`
> directory.
> 
> Alternatively, if you are on a Unix system, and have added RevBayes to your path, 
> you simply have to type `rb` in your Terminal to run the program. 
{:.instruction}
 -->
<div class="section level2">
<h2 id="overview-of-the-dispersal-extinction-cladogenesis-model">Overview of the Dispersal-Extinction-Cladogenesis Model<a class="anchor" aria-label="anchor" href="#overview-of-the-dispersal-extinction-cladogenesis-model"></a>
</h2>
<p>The Dispersal-Extinction-Cladogenesis (DEC) process models range evolution as a discrete-valued process {% cite Ree2005 Ree2008 %}. There are three key components to understanding the DEC model: range characters, anagenetic range evolution, and cladogenetic range evolution (Figure, below).</p>
<p><img src="figures/fig_range_evol_events.png" width="75%"></p>
<p>Cartoon of behavior of the DEC model.Two anagenetic events (a,b) and five cladogenetic (c–g) events are shown for a system with two areas.Areas are shaded when inhabited by a given lineage and left blank when uninhabited.Time proceeds from left to right. (a) Dispersal: a new area to be added to the species range. (b) Extirpation (or local extinction): the species range loses a previously inhabited area. (c) Narrow sympatry: When the ancestral range contains one area, both daughter lineages inherit that area. (d) Subset sympatry: When the ancestral range is widespread, one daughter inherits the ancestral range and the other daughter inherits only one area. (e) Allopatry (or vicariance): When the ancestral range is widespread, one daughter lineage inherits a subset of the ancestral areas while the other daughter inherits all remaining ancestral areas. (f) Widespread sympatry: When the ancestral range is widespread, both daughters inherit the ancestral range. (g) Jump dispersal (or founder speciation): One daughter inherits the ancestral range while the other daughter inherits a new unoccupied area.</p>
</div>
<div class="section level2">
<h2 id="discrete-range-characters">Discrete Range Characters<a class="anchor" aria-label="anchor" href="#discrete-range-characters"></a>
</h2>
<p>DEC interprets taxon ranges as presence-absence data, that is, where a species is observed or not observed across multiple discrete areas. For example, say there are three areas, A, B, and C. If a species is present in areas A and C, then its range equals AC, which can also be encoded into the length-3 bit vector, 101. Bit vectors may also be transformed into (decimal) integers, <em>e.g.</em>, the binary number 101 equals the decimal number 5.</p>
<table class="table">
<caption>Example of discrete range representations for an analysis with areas A, B, and C.</caption>
<thead><tr class="header">
<th>Range</th>
<th>Bits</th>
<th>Size</th>
<th>State</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(\emptyset\)</span></td>
<td>000</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>A</td>
<td>100</td>
<td>1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>B</td>
<td>010</td>
<td>1</td>
<td>2</td>
</tr>
<tr class="even">
<td>C</td>
<td>001</td>
<td>1</td>
<td>3</td>
</tr>
<tr class="odd">
<td>AB</td>
<td>110</td>
<td>2</td>
<td>4</td>
</tr>
<tr class="even">
<td>AC</td>
<td>101</td>
<td>2</td>
<td>5</td>
</tr>
<tr class="odd">
<td>BC</td>
<td>011</td>
<td>2</td>
<td>6</td>
</tr>
<tr class="even">
<td>ABC</td>
<td>111</td>
<td>3</td>
<td>7</td>
</tr>
</tbody>
</table>
<p>The decimal representation of range states is rarely used in discussion, but it is useful to keep in mind when considering the total number of possible ranges for a species and when processing output.</p>
<p>{% subsection Anagenetic Range Evolution | anagenetic %}</p>
<p>In the context of the DEC model, anagenesis refers to range evolution that occurs between speciation events within lineages. There are two types of anagenetic events, dispersal ({% ref dec_cartoon %}a) and (local) extinction or exitrpation ({% ref dec_cartoon %}b). Because DEC uses discrete-valued ranges, anagenesis is modeled using a continuous-time Markov chain. This, in turn, allows us to compute transition probability of a character changing from <span class="math inline">\(i\)</span> to <span class="math inline">\(j\)</span> in time <span class="math inline">\(t\)</span> through matrix exponentiation <span class="math display">\[\mathbf{P}_{ij}(t) = \left[ \exp \left\lbrace \mathbf{Q}t \right\rbrace \right]_{ij},\]</span> where <span class="math inline">\(\textbf{Q}\)</span> is the instantaneous rate matrix defining the rates of change between all pairs of characters, and <span class="math inline">\(\textbf{P}\)</span> is the transition probability rate matrix. The indices <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> represent different ranges, each of which is encoded as the set of areas occupied by the species. The probability has integrated over all possible scenarios of character transitions that could occur during <span class="math inline">\(t\)</span> so long as the chain begins in range <span class="math inline">\(i\)</span> and ends in range <span class="math inline">\(j\)</span>. We can then encode <span class="math inline">\({\bf Q}\)</span> to reflect the allowable classes of range evolution events with biologically meaningful parameters. For three areas, the rates in the anagenetic rate matrix are</p>
<p><span class="math display">\[\textbf{Q} = 
    \begin{array}{c|cccccccc}
        &amp; \emptyset &amp; A &amp; B &amp; C &amp; AB &amp; AC &amp; BC &amp; ABC \\
        \hline
        \emptyset   &amp; -     &amp; 0     &amp; 0     &amp; 0         &amp; 0         &amp; 0         &amp; 0         &amp; 0 \\
        A           &amp; e_A   &amp; -     &amp; 0     &amp; 0         &amp; d_{AB}    &amp; d_{AC}    &amp; 0         &amp; 0 \\
        B           &amp; e_B   &amp; 0     &amp; -     &amp; 0         &amp; d_{BA}    &amp; 0         &amp; d_{BC}    &amp; 0 \\
        C           &amp; e_C   &amp; 0     &amp; 0     &amp; -         &amp; 0         &amp; d_{CA}    &amp; d_{CB}    &amp; 0 \\
        AB          &amp; 0     &amp; e_B   &amp; e_A   &amp; 0         &amp; -         &amp; 0         &amp; 0         &amp; d_{AC} + d_{BC} \\
        AC          &amp; 0     &amp; e_C   &amp; 0     &amp; e_A       &amp; 0         &amp; -         &amp; 0         &amp; d_{AB} + d_{CB} \\
        BC          &amp; 0     &amp; 0     &amp; e_C   &amp; e_B       &amp; 0         &amp; 0         &amp; -         &amp; d_{BA} + d_{CA} \\
        ABC         &amp; 0     &amp; 0     &amp; 0     &amp; 0         &amp; e_C       &amp; e_B       &amp; e_A       &amp; - \\                              
    \end{array}\]</span></p>
<p>where <span class="math inline">\(e = ( e_A, e_B, e_C )\)</span> are the (local) extinction rates per area, and <span class="math inline">\(d = ( d_{AB}, d_{AC}, d_{BC}, d_{BA}, d_{CA}, d_{CB})\)</span> are the dispersal rates between areas. Notice that the sum of rates leaving the null range (<span class="math inline">\(\emptyset\)</span>) is zero, meaning any lineage that loses all areas in its range remains that way permanently.</p>
<p>To build our intuition, let’s construct a DEC rate matrix in RevBayes.</p>
<blockquote>
<p>Create a new directory on your computer called <code>RB_biogeo_tutorial</code>.</p>
<p>Navigate to your tutorial directory and execute the <code>rb</code> binary. One option for doing this is to move the <code>rb</code> executable to the tutorial directory or to create a shortcut to your executable.</p>
<p>Alternatively, if you are on a Unix system, and have added RevBayes to your path, you simply have to type <code>rb</code> in your Terminal to run the program. {:.instruction}</p>
</blockquote>
<p>Assume you have three areas</p>
<pre><code><span class="va">n_areas</span> <span class="op">&lt;-</span> <span class="fl">3</span></code></pre>
<p>First, create a matrix of dispersal rates between area pairs, with rates <span class="math inline">\(d_{AB} = d_{AC} = \ldots = d_{CB} = 1\)</span>.</p>
<pre><code><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">n_areas</span><span class="op">)</span> <span class="op">{</span>
    <span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">n_areas</span><span class="op">)</span> <span class="op">{</span>
        <span class="va">dr</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">[</span><span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1.0</span>
    <span class="op">}</span>
<span class="op">}</span></code></pre>
<p>Next, let’s create the extirpation rates with values <span class="math inline">\(e_A=e_B=e_C=1\)</span></p>
<pre><code><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">n_areas</span><span class="op">)</span> <span class="op">{</span>
    <span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">n_areas</span><span class="op">)</span> <span class="op">{</span>
        <span class="va">er</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">[</span><span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0.0</span>
    <span class="op">}</span>
    <span class="va">er</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1.0</span>
<span class="op">}</span></code></pre>
<p>When the extirpation rate matrix is a diagonal matrix (i.e. all non-diagonal entries are zero), extirpation rates are mutually independent as in (<span class="citation">@Ree2005</span>). More complex models that penalize widespread ranges that span disconnected areas are explored in later sections.</p>
<p>To continue, create the DEC rate matrix from the dispersal rates (dr) and extirpation rates (er).</p>
<pre><code><span class="va">Q_DEC</span> <span class="op">:=</span> <span class="fu">fnDECRateMatrix</span><span class="op">(</span>dispersalRates<span class="op">=</span><span class="va">dr</span>, extirpationRates<span class="op">=</span><span class="va">er</span><span class="op">)</span>
<span class="va">Q_DEC</span></code></pre>
<pre><code>    [ [ 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000 ] ,
        1.0000, -3.0000, 0.0000, 0.0000, 1.0000, 1.0000, 0.0000, 0.0000 ] ,
        1.0000, 0.0000, -3.0000, 0.0000, 1.0000, 0.0000, 1.0000, 0.0000 ] ,
        1.0000, 0.0000, 0.0000, -3.0000, 0.0000, 1.0000, 1.0000, 0.0000 ] ,
        0.0000, 1.0000, 1.0000, 0.0000, -4.0000, 0.0000, 0.0000, 2.0000 ] ,
        0.0000, 1.0000, 0.0000, 1.0000, 0.0000, -4.0000, 0.0000, 2.0000 ] ,
        0.0000, 0.0000, 1.0000, 1.0000, 0.0000, 0.0000, -4.0000, 2.0000 ] ,
        0.0000, 0.0000, 0.0000, 0.0000, 1.0000, 1.0000, 1.0000, -3.0000 ] ]</code></pre>
<p>Compute the anagenetic transition probabilities for a branch of length 0.2.</p>
<pre><code><span class="va">tp_DEC</span> <span class="op">&lt;-</span> <span class="fu">Q_DEC.getTransitionProbabilities</span><span class="op">(</span>rate<span class="op">=</span><span class="fl">0.2</span><span class="op">)</span>
<span class="va">tp_DEC</span></code></pre>
<pre><code>    [ [ 1.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000],
      [ 0.000, 0.673, 0.013, 0.013, 0.123, 0.123, 0.005, 0.050],
      [ 0.000, 0.013, 0.673, 0.013, 0.123, 0.005, 0.123, 0.050],
      [ 0.000, 0.013, 0.013, 0.673, 0.005, 0.123, 0.123, 0.050],
      [ 0.000, 0.107, 0.107, 0.004, 0.502, 0.031, 0.031, 0.218],
      [ 0.000, 0.107, 0.004, 0.107, 0.031, 0.502, 0.031, 0.218],
      [ 0.000, 0.004, 0.107, 0.107, 0.031, 0.031, 0.502, 0.218],
      [ 0.000, 0.021, 0.021, 0.021, 0.107, 0.107, 0.107, 0.616]]</code></pre>
<p>Notice how the structure of the rate matrix is reflected in the transition probability matrix. For example, ranges that are separated by multiple dispersal and extirpation events are the most improbable: transitioning from going from A to BC takes a minimum of three events and has probability 0.005.</p>
<p>Also note that the probability of entering or leaving the null range is zero. By default, the RevBayes conditions the anagenetic range evolution process on never entering the null range when computing the transition probabilities (<code>nullRange="CondSurv"</code>). This allows the model to both simulate and infer using the same transition probabilities. (Massana2015) first noted that the null range—an unobserved absorbing state—results in abnormal extirpation rate and range size estimates. Their proposed solution to eliminate the null range from the state space is enabled with the <code>nullRange="Exclude"</code> setting. The <code>nullRange="Include"</code> setting provides no special handling of the null range, and produces the raw probabilities of (Ree2005).</p>
<div class="section level3">
<h3 id="cladogenetic-range-evolution">Cladogenetic Range Evolution<a class="anchor" aria-label="anchor" href="#cladogenetic-range-evolution"></a>
</h3>
<p>The cladogenetic component of the DEC model describes evolutionary change accompanying speciation events (Below, c–g). In the context of range evolution, daughter species do not necessarily inherit their ancestral range in an identical manner. For each internal node in the reconstructed tree, one of several cladogenetic events can occur, some of which are described below.</p>
<p>Beginning with the simplest case first, suppose the range of a species is <span class="math inline">\(A\)</span> the moment before speciation occurs at an internal phylogenetic node. Since the species range is size one, both daughter lineages necessarily inherit the ancestral species range (<span class="math inline">\(A\)</span>). In DEC parlance, this is called a narrow sympatry event ({% ref dec_cartoon %}c). Now, suppose the ancestral range is <span class="math inline">\(ABC\)</span>. Under subset sympatry, one lineage identically inherits the ancestral species range, <span class="math inline">\(ABC\)</span>, while the other lineage inherits only a single area, i.e. only <span class="math inline">\(A\)</span> or <span class="math inline">\(B\)</span> or <span class="math inline">\(C\)</span> ({% ref dec_cartoon %}d). Under allopatric cladogenesis, the ancestral range is split evenly among daughter lineages, e.g. one lineage may inherit <span class="math inline">\(AB\)</span> and the other inherits <span class="math inline">\(C\)</span> ({% ref dec_cartoon %}e). For widespread sympatric cladogenesis, both lineages inherit the ancestral range, <span class="math inline">\(ABC\)</span> ({% ref dec_cartoon %}f). Finally, supposing the ancestral range is <span class="math inline">\(A\)</span>, jump dispersal cladogenesis results in one daughter lineage inheriting the ancestral range <span class="math inline">\(A\)</span>, and the other daughter lineage inheriting a previously uninhabited area, <span class="math inline">\(B\)</span> or <span class="math inline">\(C\)</span> ({% ref dec_cartoon %}g). See {% citet Matzke2012 %} for an excellent overview of the cladogenetic state transitions described in the literature (specifically see this <a href="http://phylo.wikidot.com/biogeobears#BioGeoBEARS_supermodel_graphic" class="external-link">figure</a>).</p>
<p>Make the cladogenetic probability event matrix</p>
<pre><code>clado_event_types = [ "s", "a" ]
clado_event_probs &lt;- simplex( 1, 1 )
P_DEC := fnDECCladoProbs(eventProbs=clado_event_probs,
                         eventTypes=clado_event_types,
                         numCharacters=n_areas)</code></pre>
<p><code>clado_event_types</code> defines what cladogenetic event types are used. <code>"a"</code> and <code>"s"</code> indicate allopatry and subset sympatry, as described in {% citet Ree2005 %}. Other cladogenetic events include jump dispersal <code>["j"]</code> {% cite Matzke2014 %} and full sympatry <code>["f"]</code> {% cite Landis2013a %}. The cladogenetic event probability matrix will assume that <code>eventProbs</code> and <code>eventTypes</code> share the same order.</p>
<p>Print the cladogenetic transition probabilities</p>
<pre><code><span class="va">P_DEC</span></code></pre>
<pre><code>       [
         ( 1 -&gt; 1, 1 ) = 1.0000,
         ( 2 -&gt; 2, 2 ) = 1.0000,
         ( 3 -&gt; 3, 3 ) = 1.0000,
         ...
         ( 7 -&gt; 7, 1 ) = 0.0833,
         ( 7 -&gt; 7, 2 ) = 0.0833,
         ( 7 -&gt; 7, 3 ) = 0.0833
       ]</code></pre>
<p>The cladogenetic probability matrix becomes very sparse for large numbers of areas, so only non-zero values are shown. Each row reports a triplet of states—the ancestral state and the two daughter states—with the probability associated with that event. Since these are proper probabilities, the sum of probabilities for a given ancestral state over all possible cladogenetic outcomes equals one.</p>
</div>
<div class="section level3">
<h3 id="things-to-consider">Things to Consider<a class="anchor" aria-label="anchor" href="#things-to-consider"></a>
</h3>
<p>The probabilities of anagenetic change along lineages must account for all combinations of starting states and ending states. For 3 areas, there are 8 states, and thus <span class="math inline">\(8 \times 8 = 64\)</span> probability terms for pairs of states. For cladogenetic change, we need transition probabilities for all combinations of states before cladogenesis, after cladogenesis for the left lineage, and after cladogenesis for the right lineage. Like above, for three areas, there are 8 states, and <span class="math inline">\(8 \times 8 \times 8 = 512\)</span> cladogenetic probability terms.</p>
<p>Of course, this model can be specified for more than three areas. Let’s consider what happens to the size of <strong>Q</strong> when the number of areas <span class="math inline">\(N\)</span> becomes large. For three areas, <strong>Q</strong> is of size <span class="math inline">\(8 \times 8\)</span>. For ten areas, <strong>Q</strong> is of size <span class="math inline">\(2^{10} \times 2^{10} = 1024 \times 1024\)</span>, which approaches the largest size matrices that can be exponentiated in a practical amount of time. For twenty areas, <strong>Q</strong> is size <span class="math inline">\(2^{20} \times 2^{20} \approx 10^6 \times 10^6\)</span> and exponentiation is not viable. Thus, selecting the discrete areas for a DEC analysis should be done with regard to what one hopes to learn through the analysis itself.</p>
<blockquote>
<p>Continue to the next tutorial: {% page_ref biogeo/biogeo_simple %}<br>
{:.instruction}</p>
</blockquote>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by The package maintainer.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
