<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Molecular Homology Assignment • SystematicsClass</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Molecular Homology Assignment">
<meta property="og:description" content="SystematicsClass">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">SystematicsClass</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/00_Basics_of_R/index.html">R Reminder</a>
    </li>
    <li>
      <a href="../../articles/01_homology_assignment/index.html">Morphological Homology Assignment</a>
    </li>
    <li>
      <a href="../../articles/02_molecular_homology/index.html">Molecular Homology Assignment</a>
    </li>
    <li>
      <a href="../../articles/03_massive_homology/index.html">Genome-Scale Homology</a>
    </li>
    <li>
      <a href="../../articles/04_ParsimonyAndModels/index.html">PAUP, parsimony, and nucletoide substitution models</a>
    </li>
    <li>
      <a href="../../articles/06_Bayes_and_MCMC/index.html">Introduction to MCMC using RevBayes</a>
    </li>
    <li>
      <a href="../../articles/07_molecular_data/index.html">Nucleotide substitution models</a>
    </li>
    <li>
      <a href="../../articles/08_Discrete_Morphology/index.html">Discrete morphology - Models and Tree Inference</a>
    </li>
    <li>
      <a href="../../articles/09_Partitioned_data/index.html">Partitioned data analysis</a>
    </li>
    <li>
      <a href="../../articles/10_Molecular_Clock/index.html">Tripartite Model 2 - Clock Models for Character Data</a>
    </li>
    <li>
      <a href="../../articles/11_FBD/RB_Total_Evidence_Tutorial.html">RevBayes Total Evidence Tree Estimation</a>
    </li>
    <li>
      <a href="../../articles/13_Comparative_Methods/index.html">Phylogenetic Comparative Methods for Paleobiology</a>
    </li>
    <li>
      <a href="../../articles/14_biogeography/index.html">Introduction to Phylogenetic Models of Discrete Biogeography</a>
    </li>
    <li>
      <a href="../../articles/15_Final/index.html">FinalExam</a>
    </li>
    <li>
      <a href="../../articles/LCalc/LikelihoodCalc.html">LikelihoodCalc</a>
    </li>
    <li>
      <a href="../../articles/ModelsHomework/index.html">Models Homework</a>
    </li>
    <li>
      <a href="../../articles/MolecularModelsHomework/index.html">MolecularModelsHomework</a>
    </li>
    <li>
      <a href="../../articles/MorphologyHomework/index.html">MorphologyHomework</a>
    </li>
    <li>
      <a href="../../articles/PartitionHomework/index.html">PartitionHomework</a>
    </li>
    <li>
      <a href="../../articles/PaupHomework/index.html">PAUPHomework</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Molecular Homology Assignment</h1>
            
      
      
      <div class="hidden name"><code>index.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="progressive-msa">Progressive MSA<a class="anchor" aria-label="anchor" href="#progressive-msa"></a>
</h2>
<p>Pairwise Multiple Sequence Alignment does exactly what we discussed last week, for all your sequences. Typically, this is performed by first performing a pairwise MSA, as we did above, between the two sequences with the least differences. Then, sequences are added in the order of least differences. Progressive alignments score different alignment possibilities based on the alignment and user-specified scoring matrices, such as BLOSUM 62, or others. Clustal is an example of progressive alignment, but is known to be fairly innaccurate. Below, we use T-COFFEE to get an alignment.</p>
<p>First, we will use <a href="https://tcoffee.org/Projects/tcoffee/#DOWNLOAD" class="external-link">T_COFFEE</a> to do alignments. You will find this in the software folder on your RStudio.</p>
<p>We haven’t played with any software yet in this class, so I’ll now have you create a single location for your lab materials. <code>mkdir</code> allows us to <code>m</code>a<code>k</code>e <code>dir</code>ectories. <code>cd</code> allows us to change into the directory.</p>
<pre><code>mkdir lab_one
cd lab_one</code></pre>
<p>Next, you will obtain data:</p>
<pre><code>cp ../data/sh3.fasta .</code></pre>
<p>The <code>.</code> above means “here.” It will copy data from a location to where you are.</p>
<p>We’ll now run t_coffee on its default settings (a BLOSUM62 matrix with no gap penalty). First:</p>
<pre><code>/cloud/project/software/t_coffee sh3.fasta</code></pre>
<p>Now, we’ll try a gap-opening penalty:</p>
<pre><code>/cloud/project/software/t_coffee sh3.fasta -matrix blosum62mt -gapopen 5 -outfile=gapopen5</code></pre>
<p>And one with a gap-extending penalty:</p>
<pre><code>/cloud/project/software/t_coffee sh3.fasta -matrix blosum62mt -gapopen 5 -gapext 5 outfile=gapopen5</code></pre>
<p>Try one or two more additional alignments, such as increasing the gap open or extension.</p>
<p>Now, transfer your .aln files to your personal machine.</p>
<p>Use <a href="http://tcoffee.crg.cat/apps/tcoffee/do:core" class="external-link">T-Coffee’s online viewer</a> to view them.</p>
<p>As you’re doing this, chat with a partner. What is different between the resulting alignments? Do you have a sense for which you think is “best”?</p>
</div>
<div class="section level2">
<h2 id="iterative-approaches-to-msa">Iterative Approaches to MSA<a class="anchor" aria-label="anchor" href="#iterative-approaches-to-msa"></a>
</h2>
<p>Progressive aligners make a set of assumptions, and apply those assumptions to the whole set of sequences across a phylogeny. But many datasets are fairly large, and may have different evolutionary dynamics across the tree. <em>Iterative aligners</em> use tree information to guide the process of making the alignment.</p>
<div class="figure">
<img src="fig/pasta.jpg" alt=""><p class="caption">Diagram of Sate/PASTA Algorithm</p>
</div>
<p>Now, we will download PASTA:</p>
<pre class="unix"><code>git clone https://github.com/smirarab/sate-tools-linux.git
git clone https://github.com/smirarab/pasta.git</code></pre>
<p>And we will load a couple libraries required by pasta:</p>
<pre class="unix"><code>module load java
module load python/3.5.2-anaconda-tensorflow</code></pre>
<p>Now, we build pasta:</p>
<pre class="unix"><code>python setup.py develop --user</code></pre>
<p>Change directories back into your lab_one directory. Copy the data file small.fasta from the data directory to your lab one.</p>
<p>We will use pasta to make an alignment from this small.fasta file.</p>
<pre><code>python ../software/pasta/run_pasta.py -i small.fasta</code></pre>
<p>Copy the .aln file and the .tre file to your personal machine. We’re going to look at them quickly.</p>
</div>
<div class="section level2">
<h2 id="choose-your-own-adventure-">Choose your own adventure.<a class="anchor" aria-label="anchor" href="#choose-your-own-adventure-"></a>
</h2>
<p>Below are two options to improve Pasta’s alignments. Choose one.</p>
<div class="section level3">
<h3 id="choice-of-tree-estimator">Choice of Tree Estimator<a class="anchor" aria-label="anchor" href="#choice-of-tree-estimator"></a>
</h3>
<p>Pasta allows us to build a tree for iterative estimation in different ways. <a href="http://www.microbesonline.org/fasttree/" class="external-link">FastTree</a> has some known <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0027731" class="external-link">accuracy issues</a>. That’s probably not an issue for an early stage analysis that will be improved. But let’s have a look at what happens if we use a better alignment.</p>
<p>You can change the tree estimator with the option</p>
<pre><code>--tree-estimator raxml</code></pre>
</div>
<div class="section level3">
<h3 id="changing-number-of-subsets">Changing Number of Subsets<a class="anchor" aria-label="anchor" href="#changing-number-of-subsets"></a>
</h3>
<p>Open a tree from one of the iterations in IcyTree. How many clades do you think are on this tree? What size are they? You can edit this with the option</p>
<pre><code>--max-subproblem-size=</code></pre>
<p>and choose what you think is the maximum clade size on this tree.</p>
</div>
<div class="section level3">
<h3 id="comparing-alignments">Comparing alignments<a class="anchor" aria-label="anchor" href="#comparing-alignments"></a>
</h3>
<p>As your alignments finish, copy them to your computer - the final alignments will be in the .aln file. We will view our alignment files in <a href="http://wasabiapp.org/" class="external-link">Wasabi</a>, which is a simple-browser based alignment.</p>
<p>It can be very hard to appreciate the differences between alignments by eye. We will try making comparisons with <a href="https://github.com/smirarab/FastSP" class="external-link">FastSP</a>, which gives <a href="https://academic.oup.com/bioinformatics/article/27/23/3250/234345" class="external-link">some at-a-glance comparisons</a>.</p>
<p>Change directories back to your home directory. Clone the FastSP software:</p>
<pre><code>git clone https://github.com/smirarab/FastSP.git</code></pre>
<p>Now, change back to your previous directory with:</p>
<pre><code>cd -</code></pre>
<p>You can call FastSP like so:</p>
<pre><code>java -jar /cloud/project/software/FastSP/FastSP.jar -r reference_alignment_file -e estimated_alignment_file</code></pre>
<p>So, for example, if I wanted to compare the alignment I estimated with Pasta, using FastTree, and the one with Pasta and Raxml, my command would look like</p>
<pre><code>java -jar /cloud/project/software/FastSP/FastSP.jar -r pastajob.marker001.small.aln -e pastajob1.marker001.small.aln</code></pre>
<p>FastSP calculates a few summary statistics on alignments. The output should look like:</p>
<pre><code>Reference alignment: /Users/april/Documents/SELUSys2018/lab1/output/pasta_exercise2/smallrax2.marker001.small.aln ...
Estimated alignment: /Users/april/Documents/SELUSys2018/lab1/output/pasta_exercise3/subset_exp.marker001.small.aln ...
MaxLenNoGap= 1027, NumSeq= 32, LenRef= 1169, LenEst= 1180, Cells= 75168
computing ...
Number of shared homologies: 447050
Number of homologies in the reference alignment: 486886
Number of homologies in the estimated alignment: 487792
Number of correctly aligned columns: 716
Number of aligned columns in ref. alignment: 1069
SP-Score 0.9181820795833111
Modeler 0.9164766949847476
SPFN 0.0818179204166889
SPFP 0.0835233050152524
Compression 1.009409751924722
TC 0.6697848456501403</code></pre>
<p>The first line is some information that shouldn’t surprise us too much: the maximum length of sequence in the file without gaps, NumSeq is the number of sequences in the file, the two Lens are the lengths of the alignments (including gaps), and cells are the total number of cells (num seq x number of nucleotides per line, summed for both alignments).</p>
<p>Number of shared homologies refers to how many filled cells are filled in both, or gaps in both. Number of homologies in the reference/estimated alignment is the number of pairs of letters from the input sequences that occur in the same site - this can be larger than the amount of cells, because you’ll make multiple comparisons per column.</p>
<p>Based on these metrics, which alignment do you prefer? What other information would you want to know before choosing an alignment for your project?</p>
</div>
</div>
<div class="section level2">
<h2 id="coestimation-of-msa-and-phylogeny">Coestimation of MSA and Phylogeny<a class="anchor" aria-label="anchor" href="#coestimation-of-msa-and-phylogeny"></a>
</h2>
<p>Phylogenetic esitmation generally assumes that we know the alignment without error. We’ve already seen in our examples cases where we do not get the same alignment between methods. This is particularly true in areas that are hard to align. For example, in the below paper, they estimate a tree for fungi, which are deeply-diverged, and very diverse. The alignment has many problematic regions. What they showed in this paper was that by not accounting for the alignment uncertainty, support was overestimated for their tree hypothesis.</p>
<div class="figure">
<img src="fig/Lutzoni.png" alt=""><p class="caption">Lutzoni paper</p>
</div>
<p>We haven’t talked about Bayesian estimation, so I’m going to say very little on joint estimation of alignment and topology. This is a method that allows for a wide range of models to be deployed in order to estimate both the alignment and the tree, and shows great promise for difficult alignment issues. We have a few floating open labs throughout the semester, so if this is a topic of interest to people, we can revisit it then.</p>
<p>The primary software that performs this analysis is <a href="http://www.bali-phy.org/Tutorial3.html" class="external-link">Bali-Phy</a>, which is described <a href="http://www.bali-phy.org/Redelings_and_Suchard_2005.pdf" class="external-link">here</a>.</p>
</div>
<div class="section level2">
<h2 id="alignment-homework">Alignment Homework<a class="anchor" aria-label="anchor" href="#alignment-homework"></a>
</h2>
<div class="section level3">
<h3 id="some-of-these-questions-do-not-have-a-right-answer-">Some of these questions do not have a right answer.<a class="anchor" aria-label="anchor" href="#some-of-these-questions-do-not-have-a-right-answer-"></a>
</h3>
<p>Feel free to work in groups, and discuss the assignments as needed. However, I do expect you to turn in your own copy, with answers in your own words.</p>
<ol style="list-style-type: decimal">
<li><p>Some algorithms treat a gap as a single penalty value, regardless of how large the gap is. Others assess a gap opening penalty, then a smaller gap extension penalty. When (i.e. what kind of biological scenarios) might you think it might be better to use one algorithm over the other?</p></li>
<li><p>Breaking problems into subproblems is a common way to attack a tough problem. In the case of iterative alignments, we break the tree into smaller pieces. In assembling our ddRAD matrices, we build stacks of loci per sample first. Are there biological questions for which you expect this would not be helpful?</p></li>
</ol>
</div>
</div>
<div class="section level2">
<h2 id="references">References:<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>T-Coffee Manual: <a href="http://tcoffee.readthedocs.io/en/latest/" class="external-link uri">http://tcoffee.readthedocs.io/en/latest/</a>
</li>
<li>Pasta Algorithm Description: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4424971/" class="external-link uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4424971/</a>
</li>
<li>Sate Algorithm Description: <a href="https://academic.oup.com/sysbio/article/61/1/90/1680002" class="external-link uri">https://academic.oup.com/sysbio/article/61/1/90/1680002</a>
</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by The package maintainer.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
