<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>PAUP, parsimony, and nucletoide substitution models • SystematicsClass</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="PAUP, parsimony, and nucletoide substitution models">
<meta property="og:description" content="SystematicsClass">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">SystematicsClass</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/00_Basics_of_R/index.html">R Reminder</a>
    </li>
    <li>
      <a href="../../articles/01_homology_assignment/index.html">Morphological Homology Assignment</a>
    </li>
    <li>
      <a href="../../articles/02_molecular_homology/index.html">Molecular Homology Assignment</a>
    </li>
    <li>
      <a href="../../articles/03_massive_homology/index.html">Genome-Scale Homology</a>
    </li>
    <li>
      <a href="../../articles/04_ParsimonyAndModels/index.html">PAUP, parsimony, and nucletoide substitution models</a>
    </li>
    <li>
      <a href="../../articles/06_Bayes_and_MCMC/index.html">Introduction to MCMC using RevBayes</a>
    </li>
    <li>
      <a href="../../articles/07_molecular_data/index.html">Nucleotide substitution models</a>
    </li>
    <li>
      <a href="../../articles/08_Discrete_Morphology/index.html">Discrete morphology - Models and Tree Inference</a>
    </li>
    <li>
      <a href="../../articles/09_Partitioned_data/index.html">Partitioned data analysis</a>
    </li>
    <li>
      <a href="../../articles/10_Molecular_Clock/index.html">Tripartite Model 2 - Clock Models for Character Data</a>
    </li>
    <li>
      <a href="../../articles/11_FBD/RB_Total_Evidence_Tutorial.html">RevBayes Total Evidence Tree Estimation</a>
    </li>
    <li>
      <a href="../../articles/13_Comparative_Methods/index.html">Phylogenetic Comparative Methods for Paleobiology</a>
    </li>
    <li>
      <a href="../../articles/14_biogeography/index.html">Introduction to Phylogenetic Models of Discrete Biogeography</a>
    </li>
    <li>
      <a href="../../articles/15_Final/index.html">FinalExam</a>
    </li>
    <li>
      <a href="../../articles/LCalc/LikelihoodCalc.html">LikelihoodCalc</a>
    </li>
    <li>
      <a href="../../articles/ModelsHomework/index.html">Models Homework</a>
    </li>
    <li>
      <a href="../../articles/MolecularModelsHomework/index.html">MolecularModelsHomework</a>
    </li>
    <li>
      <a href="../../articles/MorphologyHomework/index.html">MorphologyHomework</a>
    </li>
    <li>
      <a href="../../articles/PartitionHomework/index.html">PartitionHomework</a>
    </li>
    <li>
      <a href="../../articles/PaupHomework/index.html">PAUPHomework</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>PAUP, parsimony, and nucletoide substitution models</h1>
            
      
      
      <div class="hidden name"><code>index.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="maximum-parsimony-analysis-using-paup">Maximum Parsimony Analysis Using PAUP<a class="anchor" aria-label="anchor" href="#maximum-parsimony-analysis-using-paup"></a>
</h2>
<p>Today, we will use PAUP* (Phylogenetic Analysis Using Parsimony [*and other methods]) to estimate a phylogeny using the maximum parsimony optimality criterion. We will also practice performing bootstrap and jackknife quantification of uncertainty. In the process, we will also see some of the basics of how phylogenetic software will read data and understand models.</p>
</div>
<div class="section level2">
<h2 id="download-paup-and-data">Download PAUP and Data<a class="anchor" aria-label="anchor" href="#download-paup-and-data"></a>
</h2>
<p>I have already installed paup for you. But if you wanted to, you could do it like so.</p>
<pre class="unix"><code> curl -O  http://phylosolutions.com/paup-test/paup4a168_centos64.gz
 gunzip paup4a168_centos64.gz 
</code></pre>
<p>Now, move into your lab_three directory. Make a Data, Results, and Scripts subdirectories inside of it. Enter the Data directory. Copy the data from class repository (<code>data/PAUPlab</code>). The following is a fill in the blank - figure out how you get from your data directory to the classroom data dir.</p>
<pre class="unix"><code>
cp __/data/PAUPlab ./Data</code></pre>
</div>
<div class="section level2">
<h2 id="paup">PAUP<a class="anchor" aria-label="anchor" href="#paup"></a>
</h2>
<p>Paup is one of the older phylogenetics software packages available, and Dave Swofford has put considerable effort into ensuring that the main mechanisms and models are correct. Many of us who do other software development use PAUP as a check on our own calculations.</p>
<p>PAUP reads in data in the Nexus format. Go ahead and use nano to open the primate dataset.</p>
<p>Next, we will open PAUP:</p>
<pre class="unix"><code><span class="va">.</span><span class="op">/</span><span class="va">paup4a168_centos64</span></code></pre>
<p>You should get something like the below:</p>
<pre class="unix"><code>P A U P *
Portable version 4.0b10 for Unix
Sat Feb 17 16:04:16 2018

      -----------------------------NOTICE-----------------------------
        This is a beta-test version.  Please report any crashes,
        apparent calculation errors, or other anomalous results.
        There are no restrictions on publication of results obtained
        with this version, but you should check the WWW site
        frequently for bug announcements and/or updated versions.  
        See the README file on the distribution media for details.
      ----------------------------------------------------------------

paup&gt;
</code></pre>
<p>Next, we will load in the data file, using the execute command. Execute is kind of PAUP’s catch-all for loading in data.</p>
<pre class="unix"><code>execute pauplab/primate-mtDNA.nex</code></pre>
<p>Let’s have a look at the screen output. What information does it tell us?</p>
<p>Some other useful commands are:</p>
<pre><code>cstatus
tstatus
showmatrix
showdist
log file="mylogfile"</code></pre>
<p>Try each command. What is it telling you? If any of them are still unclear, you can add a question mark after the command to get more info.</p>
<p>First, we will compute a parsimony tree using the command all trees:</p>
<pre class="unix"><code><span class="va">alltrees</span>;</code></pre>
<p>This will take a moment. Let’s talk about parsimony, and the alltrees command. What is that command doing? What is parsimony doing? [go to board exercise]</p>
<p>Look at the best tree with the command showtree:</p>
<pre class="unix"><code><span class="va">showtree</span>;</code></pre>
<p>Are we surprised by the length of the tree, given the number of characters?</p>
<p>Next, we will try heuristic searching:</p>
<pre class="unix"><code><span class="va">hsearch</span></code></pre>
<p>Note that a heuristic is being applied. What is a heuristic?</p>
<p>Did it make a difference to the score of the best tree? Why?</p>
<p>Let’s try some data that are too large for an exhaustive search:</p>
<pre><code>execute BardenClean.nex</code></pre>
<p>This is some of my research data. Try to execute alltrees now. What does this message mean? Now try the heuristic search.</p>
<p>What is the score of the best tree you found _________. How many best trees were there? ________ Try one of the alternative swap measures (either NNI or SPR).</p>
<pre><code>hsearch swap= ______</code></pre>
<p>What was the score of your best tree? When would you expect the swap algorithm to matter strongly?</p>
<p>Now, save some trees:</p>
<pre class="unix"><code>savetrees from=1 to=1 file=Results/tree1.tre;
savetrees from=2 to=2 file=Results/tree1.tre;</code></pre>
<p>Download them. How are these trees different? Why is PAUP saving multiple trees?</p>
<p>To confirm our intuition, let’s score them all under parsimony:</p>
<pre><code>pscore 1-121</code></pre>
<p>This will print the number of parsimony steps on each tree. Does this confirm what you thought before?</p>
<p>We can summarize the trees with a consensus tree:</p>
<pre><code>contree all / treefile=Results/contree.tre;</code></pre>
<p>If you look at the help for the consensus tree function, you see that there are several options for how to compute the consensus tree. The default is <strong>strict:</strong> the tree which contains only groups found in all trees. <strong>Semistrict</strong> is similar, containing all compatible groups. This tree may contain clades that are not present in all trees, but are not contradicted by them either. <strong>Majority Rule</strong> is a common display setting, showing groups that are present in more than 50% of trees. An <strong>Adams Consensus Tree</strong> will place taxa that vary in their placement in the most inclusive sport within the group.</p>
<p>Try a couple of these tree types. How do they differ in terms of their tree representation?</p>
</div>
<div class="section level2">
<h2 id="the-bootstrap">The Bootstrap<a class="anchor" aria-label="anchor" href="#the-bootstrap"></a>
</h2>
<p>We will now bootstrap our data. The bootstrap analysis generates a number of matrices (in our case, 100) from the original data matrix. Every taxon is represented in every matrix. Columns, however, are drawn, with replacement up until the original size of the matrix is obtained (i.e., the matrices are resampled with replacement). This process randomizes which characters are in the matrix, and measures repeatability. Typically more than 100 bootstraps will be run, and will generate support values for a node that run from 0-100.</p>
<p>Reload the primate data. Now, begin a bootstrap run:</p>
<pre class="unix"><code>bootstrap nreps=100 search=heuristic /addseq=random;</code></pre>
<p>Save the tree:</p>
<pre class="unix"><code>savetrees from=1 to=1 savebootp=nodelabels file=output/bootstrap.tre;</code></pre>
</div>
<div class="section level2">
<h2 id="the-jackknife">The Jackknife<a class="anchor" aria-label="anchor" href="#the-jackknife"></a>
</h2>
<p>The jackknife is similar to the bootstrap, but is performed without replacement. Therefore, each dataset is smaller.</p>
<pre class="unix"><code>jackknife nreps=100 search=heuristic /addseq=random;</code></pre>
<p>Why would this be useful?</p>
<pre class="unix"><code>savetrees from=1 to=1 savebootp=nodelabels file=output/jackknife.tre;</code></pre>
</div>
<div class="section level2">
<h2 id="what-is-parsimony-doing-anyway">What is parsimony doing anyway?<a class="anchor" aria-label="anchor" href="#what-is-parsimony-doing-anyway"></a>
</h2>
<p>What assumption does parsimony make about how characters evolve?</p>
<p>Go ahead, log into LONI, start an interactive sessions and load paup.</p>
<p>Now, we’ll start paup and read in the data.</p>
<pre class="unix"><code>paup

execute Data/primate-mtDNA.nex

alltrees</code></pre>
<p>Copy down the score of the best tree.</p>
<p>What about making different assumptions?</p>
<p>Exit Paup.</p>
<p>Copy the BardenClean.nex into our Data folder. Open the file. What are these data? Please add this matrix to the end of the file:</p>
<pre><code>begin assumptions;
 usertype my_ctype stepmatrix=2
     0 1
 [0] . 1
 [1] 0 .

 ;
end;
</code></pre>
<p>What is this matrix specifying? Save it. Now, restart PAUP and do a quick hsearch:</p>
<pre class="unix"><code>execute Data/BardenClean.nex
hsearch</code></pre>
<p>Why is the score on this so much less than the multistate data set?</p>
<p>Now, let’s try using our custom matrix:</p>
<pre><code>ctype my_ctype:1
hsearch</code></pre>
<p>We have imposed a new step matrix on the first character in the matrix.</p>
<p>Let’s draw it on the board. What kind of data do you think would fit this data? What kind of data would fit parsimony?</p>
<p>How does this impact the parsimony score of the data? Why does this make sense? What does this mean for how we conceive of these models?</p>
</div>
<div class="section level2">
<h2 id="modeling-sequence-evolution">Modeling Sequence Evolution<a class="anchor" aria-label="anchor" href="#modeling-sequence-evolution"></a>
</h2>
<p>Load in the primate-mtDNA.nex file. What does a step matrix look like for parsimony for nucleotide sequence data?</p>
<p>Perform an exhaustive parsimony search. Build a consensus tree and save it.</p>
<p>Let’s talk about different assumptions we can make about the way molecular sequences evolve.</p>
<div class="section level3">
<h3 id="transitions-and-transversions">Transitions and Transversions<a class="anchor" aria-label="anchor" href="#transitions-and-transversions"></a>
</h3>
<div class="figure">
<img src="../fig/Q.png" alt=""><p class="caption">Image of Transitions and Transversions via Wikipedia</p>
</div>
<p>Exercise on board making this transition matrix, then in text editor.</p>
<p>Apply the new matrix to the whole primate dataset, and perform an exhaustive parsimony search.</p>
<p>How has this changed the parismony score of the best tree found? Make a consensus tree and save it. Download and view the two trees we’ve made. Are they different?</p>
<p><strong>Homework</strong>: Try the same analysis, except with a 3-1 weighting on transitions to transversions.</p>
</div>
<div class="section level3">
<h3 id="base-frequencies">Base Frequencies<a class="anchor" aria-label="anchor" href="#base-frequencies"></a>
</h3>
<p>This is where we will transition from thinking about parsimony to thinking about other methods. Think about a dataset that you have. Do you think that nucleotide bases occur at equal frequencies in those data?</p>
<p>On board: show how the probability of observing a change changes as a function of base frequency.</p>
</div>
</div>
<div class="section level2">
<h2 id="so-what-is-likelihood-anyway">So what is likelihood anyway?<a class="anchor" aria-label="anchor" href="#so-what-is-likelihood-anyway"></a>
</h2>
<p>First, let’s review probability. Dice rolling activity.</p>
<p>Next, let’s look at how probability applies to phylogenetics. On board: tree, with ancestral states and probabilities labeled on them.</p>
<p>Now, let’s work out the probability of this sequence under JC69.</p>
<p>Now, let’s do this in paup.</p>
<pre class="unix"><code>set crit=like;
lset nst=1 basefreq=equal;
hsearch;</code></pre>
<p>But … wait, we just calculated a single site likelihood, and it was really small. Why is this number so big?</p>
<p>Now let’s try with two transition rates:</p>
<pre class="unix"><code>set crit=like;
lset nst=2 basefreq=equal;
hsearch;</code></pre>
<p>What is your likelihood score? Do you think this is a better or a worse score?</p>
<p>How does the two previous commands differ from:</p>
<pre class="unix"><code>set crit=like;
lset nst=1 basefreq=emp;
hsearch;</code></pre>
<p>What is the score of this model? And finally:</p>
<pre class="unix"><code>set crit=like;
lset nst=2 basefreq=emp;
hsearch;</code></pre>
<p>What is the score of this model?</p>
<p><strong>Homework</strong>:</p>
<p>I am an arrogant jerk, and I would like you to <a href="https://github.com/graemetlloyd/ProjectWhalehead/blob/master/vignettes/MS.pdf" class="external-link">read a paper I have accepted</a> right now. Write a note for me if any of my arguments in the paper change your mind on the below questions.</p>
<ol style="list-style-type: decimal">
<li><p>We talked about heuristics Monday. Will a heuristic always find you the best tree? How are you, personally, going to make peace with this in your work?</p></li>
<li><p>One of the cases that Graeme and I make in the paper I made you read because I’m an arrogant jerk is that there are often a lot of trees that are plausible solutions for a particular analysis, and that researchers should consider multiple possible topologies when making inferences using trees. For you particular research question, do you think this will be an issue? How would you need to modify your dissertation analyses to accommodate multiple possible topologies?</p></li>
<li><p>In your own words, why do base frequencies matter?</p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by The package maintainer.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
